mesonbuild/modules/pkgconfig.py

```diff
@@ -136,6 +138,26 @@ def generate(self, state, args, kwargs):
         pub_reqs = mesonlib.stringlistify(kwargs.get('requires', []))
         priv_reqs = mesonlib.stringlistify(kwargs.get('requires_private', []))
         conflicts = mesonlib.stringlistify(kwargs.get('conflicts', []))
+
+        def parse_variable_list(stringlist):
+            reserved = ['prefix', 'libdir', 'includedir']
+            variables = []
+            for var in stringlist:
+                try:
+                    k, v = var.split('=')
```

Comment:
does avoiding the exception give us any benefit here? if we get it, we error out anyway so optimization here shouldn't be the key. and for easier comparison and conversion into tuples I need both sides of the =. Otherwise, it'd be a loop with "if foo.startswith("{}=".format(k))" which is a lot uglier than "if k in reserved"